src/geometry/euclidean/ceva_theorem.lean

```diff
@@ -0,0 +1,325 @@
+  { apply vadd_right_cancel (S.coord 2 O • (D -ᵥ S.points 2)),
+    simp only [vsub_eq_sub, sub_add_cancel, vadd_eq_add],
+    simp only [fin.sum_univ_succ, fin.sum_univ_zero, add_zero] at h,
+    convert h using 1,
+    norm_num,
+    abel },
+  have hO : O ∈ ({D, O, S.points 2} : set V),
+  { simp only [set.mem_insert_iff, true_or, eq_self_iff_true, or_true] },
+  rw (collinear_iff_of_mem ℝ hO) at h₂,
+  cases h₂ with v₁ hv₁,
+  have hD₁ : D ∈ ({S.points 0, S.points 1, D} : set V),
+  { simp only [set.mem_insert_iff, set.mem_singleton, or_true] },
+  rw (collinear_iff_of_mem ℝ hD₁) at h₁,
+  cases h₁ with v₂ hv₂,
+  obtain ⟨r₂, hC₁⟩ := hv₁ (S.points 2)
+    (by simp only [set.mem_insert_iff, set.mem_singleton, or_true]),
+  obtain ⟨r₃, hA₁⟩ := hv₂ (S.points 0)
+    (by simp only [set.mem_insert_iff, true_or, eq_self_iff_true]),
+  obtain ⟨r₄, hB₁⟩ := hv₂ (S.points 1)
+    (by simp only [set.mem_insert_iff, true_or, eq_self_iff_true, or_true]),
+  obtain ⟨r₁, hD₁⟩ := hv₁ D
+    (by simp only [set.mem_insert_iff, true_or, eq_self_iff_true]),
+  simp only [hC₁, hA₁, hB₁, hD₁, vadd_vsub_vadd_cancel_right, vadd_vsub, vsub_vadd_eq_vsub_sub,
+    zero_sub, smul_neg, sub_self] at hsub,
+  rw [hB₁, hA₁, hD₁],
+  simp only [vadd_vsub_vadd_cancel_right, vadd_vsub, vsub_vadd_eq_vsub_sub, zero_sub, smul_neg,
+   sub_self],
+  have hlinind := affine_basis.pair_lin_indep S hA₁ hB₁ hC₁ hD₁,
+  have hv₁ : (r₁ + S.coord 2 O • r₂ - S.coord 2 O • r₁) • v₁ = r₁ • v₁ -
+    S.coord 2 O • (r₁ • v₁ - r₂ • v₁),
+  rw ← sub_smul,
+  rw ← smul_assoc,
+  simp only [smul_eq_mul, mul_sub, sub_smul, add_smul],
+  rw sub_sub_assoc_swap,
+  have hv₂ : (- S.coord 0 O • r₃ - S.coord 1 O • r₄) • v₂ = -(S.coord 0 O • r₃ • v₂) +
+    -(S.coord 1 O • r₄ • v₂),
+  simp only [sub_smul, ← smul_assoc, smul_eq_mul, neg_smul, ← sub_eq_add_neg],
+  have h₂ : (r₁ + S.coord 2 O • r₂ - S.coord 2 O • r₁) • v₁ = (- S.coord 0 O • r₃ -
+    S.coord 1 O • r₄) • v₂,
+  rw hv₁,
+  rw hv₂,
+  exact hsub,
+  simp only [smul_eq_mul] at h₂,
+  rw [← sub_eq_add_neg, ← neg_smul, ← smul_assoc, ← smul_assoc, ← sub_smul],
+  by_cases h₃ : (r₁ + S.coord 2 O * r₂ - S.coord 2 O * r₁) = 0,
+  simp only [h₃, zero_smul] at h₂,
+  simp only [smul_eq_mul, ← h₂],
+  rw [← eq_inv_smul_iff₀, ← smul_assoc] at h₂,
+  rw linear_independent_fin2 at hlinind,
+  cases hlinind with h₄ h₅,
+  specialize h₅
+    ((r₁ + S.coord 2 O * r₂ - S.coord 2 O * r₁)⁻¹ • (-S.coord 0 O * r₃ - S.coord 1 O * r₄)),
+  simp only [neg_mul_eq_neg_mul_symm, matrix.head_cons, smul_eq_mul, ne.def,
+    matrix.cons_val_one, matrix.cons_val_zero] at h₅,
+  exfalso,
+  apply h₅,
+  rw [h₂, smul_eq_mul, neg_mul_eq_neg_mul],
+  exact h₃
+end
+
+lemma affine_basis.interior_coord_pos
+  {ι : Type*} [fintype ι] {O : V} {T : affine_basis ι ℝ V}
+  (h : O ∈ interior (convex_hull ℝ (set.range T.points))) {i : ι} : 0 < T.coord i O :=
+begin
+  rw interior_convex_hull_aff_basis T at h,
+  rw set.mem_set_of_eq at h,
+  exact h i,
+end
```

Comment:
I think this lemma belongs alongside the existing `interior_convex_hull_aff_basis` that it uses.